gcfgenv.go

```diff
@@ -86,13 +86,23 @@ func readWithMapInto(r io.Reader, env map[string]string, prefix string, config i
 	return err
 }
 
+func gcfgTagOrStructName(field reflect.StructField) string {
+	t := field.Tag.Get("gcfg")
+	if t != "" {
+		// also need to replace dashes with underscores
+		// field.Name uses this convention
+		return strings.ReplaceAll(t, "-", "_")
+	}
+	return field.Name
+}
+
 func setGcfgWithEnvMap(ref reflect.Value, prefix string, env map[string]string) error {
 	refType := ref.Type()
 	for i := 0; i < refType.NumField(); i++ {
 		sec := ref.Field(i)
 		secStructField := refType.Field(i)
 		secType := sec.Type()
-		secPrefix := prefix + strings.ToUpper(secStructField.Name)
+		secPrefix := prefix + strings.ToUpper(gcfgTagOrStructName(secStructField))
```

Comment:
Since this utility function is always wrapped in `strings.ToUpper()`, why not make that part of the function itself?